xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/ListClass.java

```diff
@@ -742,7 +742,7 @@ public void displayView(StringBuffer buffer, String name, String prefix, BaseCol
             }
             buffer.append(StringUtils.join(newlist, separator));
         } else {
-            buffer.append(getDisplayValue(prop.getValue(), name, map, context));
+            buffer.append(XMLUtils.escape(getDisplayValue(prop.getValue(), name, map, context)));
```

Comment:
How come the XML escaping is needed only here and not above? I'm asking because the ``newlist`` above is also built with ``getDisplayValue()``. This lack of consistency needs a comment at least.

Reply:
IIRC it is related to https://jira.xwiki.org/browse/XWIKI-9680 and adding an escape no line 743 makes `StaticListClassTest#testDisplayView` fail. But I am not sure to understand the previous decisions. I'll dig deeper to understand what's going on.

Reply:
As I understand it, fixing XWIKI-17374 and XWIKI-9680 are mutually exclusive.